material-stepper/src/main/java/com/stepstone/stepper/StepperLayout.java

```diff
@@ -496,6 +509,11 @@ private void onNext() {
         if (verifyCurrentStep(step)) {
             return;
         }
+
+        //if moving forward and got no errors, set hasError to false, so we can have the tab with the check mark.
+        if(mStepperType instanceof TabsStepperType)
```

Comment:
We could replace the if statements with polymorphism i.e. this could be changed to 

    mStepperType.setErrorStep(mCurrentStepPosition, flag)

by default this would do nothing, but for TabsStepperType this would call:

    mTabsContainer.setErrorStep(stepPosition, false);